pkg/astnormalization/inject_input_default_values.go

```diff
@@ -0,0 +1,115 @@
+package astnormalization
+
+import (
+	"github.com/buger/jsonparser"
+	"github.com/jensneuse/graphql-go-tools/pkg/ast"
+	"github.com/jensneuse/graphql-go-tools/pkg/astvisitor"
+)
+
+func injectInputFieldDefaults(walker *astvisitor.Walker) *inputFieldDefaultInjectionVisitor {
+	visitor := &inputFieldDefaultInjectionVisitor{
+		Walker:       walker,
+		variableTree: make([]string, 0),
+	}
+	walker.RegisterEnterDocumentVisitor(visitor)
+	walker.RegisterVariableDefinitionVisitor(visitor)
+	return visitor
+}
+
+type inputFieldDefaultInjectionVisitor struct {
+	*astvisitor.Walker
+
+	operation  *ast.Document
+	definition *ast.Document
+
+	variableName string
+	variableTree []string
+}
+
+func (in *inputFieldDefaultInjectionVisitor) EnterDocument(operation, definition *ast.Document) {
+	in.operation, in.definition = operation, definition
+}
+
+func (in *inputFieldDefaultInjectionVisitor) EnterVariableDefinition(ref int) {
+	in.variableName = in.operation.VariableDefinitionNameString(ref)
+
+	exists, err := in.variableKeyExists(in.variableName)
+	if err != nil {
+		in.StopWithInternalErr(err)
+		return
+	}
+	if !exists {
+		return
+	}
+	typeName := in.operation.BaseTypeNameBytes(ref)
+	node, found := in.definition.Index.FirstNodeByNameBytes(typeName)
+	if !found {
+		return
+	}
+	if node.Kind != ast.NodeKindInputObjectTypeDefinition {
+		return
+	}
+
+	in.recursiveInjectInputFields(node.Ref, in.variableName)
+}
+
+func (in *inputFieldDefaultInjectionVisitor) recursiveInjectInputFields(inputObjectRef int, fieldName string) {
```

Comment:
maybe call receiver `i` or `v`?

`in.Smth` - for my IMHO is a bad name for receiver cause it looks like `input` - `in`